### PR TITLE
Update SmoldotProvider and respective tests to support smoldot v3

### PR DESCRIPTION
**Pull request by wirednkod**
- Bump version of smoldot to v3.1
- Update SmoldotProvider to new API; 
- Update smoldotTestUtilScripts;

(Tests fail on SmoldotMediator - for now opening a draft PR)

### CODE REVIEW

File: packages/connect/package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "eventemitter3": "^4.0.7",
     "file-entry-cache": "^6.0.1",
     "mkdirp": "^1.0.4",
-    "smoldot": "0.2.9"
+    "smoldot": "0.3.1"
```
Comment (tomaka) [inline packages/connect/package.json:42]:
```suggestion
    "smoldot": "0.3.2"
```

Just published 0.3.2

File: yarn.lock [hunk 2]
```diff
@@ -13252,10 +13252,20 @@ slice-ansi@^4.0.0:
...
+  dependencies:
+    buffer "^6.0.1"
+    performance-now "^2.1.0"
+    randombytes "^2.1.0"
+    websocket "^1.0.32"
+
+smoldot@0.3.1:
+  version "0.3.1"
+  resolved "https://registry.yarnpkg.com/smoldot/-/smoldot-0.3.1.tgz#fbf135db18e0723e8b3e7f94ef9974bd82c7a0a0"
+  integrity sha512-Uq5n0fS9lq3LDsbFlnvFd7ujC6TzEn0jtcInpuIijQznY0s8tfED65Z8BSHGfMJh+AC4oKAB1eIE0BJYRJBo8Q==
```
Comment (tomaka) [inline yarn.lock:13268]: Well, this need to be updated too.

Comment (wirednkod) [reply yarn.lock:13268]: done

File: packages/smoldot-test-utils/src/index.ts [hunk 3]
```diff
@@ -164,23 +168,24 @@ export const respondWith = (jsonResponses: string[]) => {
  * first returning the subscription id.  The second a response to the
  * subscription.
  */
-const createRequestProcessor = (options: SmoldotOptions, responder: RpcResponder, healthResponder: RpcResponder) => {
-  return (rpcRequest: string, chainIndex: number) => {
+const createRequestProcessor = (options: SmoldotAddChainOptions, responder: RpcResponder, healthResponder: RpcResponder) => {
+  return (rpcRequest: string) => {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
```
Comment (raoulmillais) [inline packages/smoldot-test-utils/src/index.ts:173]: We should review all the lint disable statements when we remove the temporary mock for < 0.3.x

I'm suspicious of the need for this one too.

**Review by raoulmillais — APPROVED** [review]
The temp mock and spy are :nauseated_face: but they'll be very shortlived :crossed_fingers:

**Comment by wirednkod** [comment]
> The temp mock and spy are :nauseated_face: but they'll be very shortlived :crossed_fingers:

Well they are, but still a copy paste from previous version. They will be shortlived though. As discussed this will have to be discontinued as it wont support (along w smoldot v2.0) the nodes for long

**Pull request merged**